app/backend/src/clj/teet/activity/activity_db.clj

```diff
@@ -181,3 +182,20 @@
        (-> (du/entity db activity-id)
            :activity/name :db/ident)
        #{}))
+
+(defn activity-task-has-files?
+  "Check a task under activity has undeleted files."
+  [db activity-id]
+  (seq
+    (filter boolean
+      (map (partial file-db/task-has-files? db)
+        (mapv
+          (comp :db/id first)
+          ;; Find all Tasks that belong to Activity
+          (d/q '[:find (pull ?t [:db/id])
```

Comment:
You don't need the `(pull ?t [:db/id])` just having `?t` in it's place returns just the `:db/id`